.github/workflows/main.yaml

```diff
@@ -0,0 +1,170 @@
+# Reusable Cobalt CI workflow.
+
+name: main
+
+on:
+  workflow_call:
+    inputs:
+      platform:
+        description: 'Cobalt platform.'
+        required: true
+        type: string
+
+# Global env vars.
+env:
+  REGISTRY: ghcr.io
+  IPV6_AVAILABLE: 0
+  IS_BUILDBOT_DOCKER: 1
+  #BUILD_ID_SERVER_URL:
+  IS_CI: 1
+  IS_DOCKER: 1
+  NINJA_STATUS: '[%e sec | %f/%t %u remaining | %c/sec | j%r]'
+  SCCACHE: 1
+  SCCACHE_GCS_BUCKET: cobalt-actions-devel-sccache-linux
+  SCCACHE_GCS_OAUTH_URL: http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token
+  SCCACHE_GCS_RW_MODE: READ_WRITE
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Builds, tags, and pushes Cobalt docker images to ghr.
+  docker-image:
+    permissions:
+      packages: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout files
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2 
+      - name: Get docker file changes
+        id: changed-files
+        uses: tj-actions/changed-files@v13.1
+        with:
+          files: |
+            docker-compose.yml
+            docker/**
+      - id: get-docker-service
+        run: |
+          export docker_service=$(cat ${GITHUB_WORKSPACE}/.github/config/${{ inputs.platform }}.json | jq -r '.docker_service')
+          echo "::set-output name=docker_service::$docker_service"
+          echo "docker_service=$docker_service" >> $GITHUB_ENV
```

Comment:
What does this do?

Reply:
It extracts 'docker_service' parameter from the json config file which points to a service to build in docker-compose file (e.g. build-linux, build-evergreen, etc.).  It then sets env var with same name to be used in next steps, and sets an output parameter with same name to be used in build jobs (tells build jobs what docker image to build in).